WordPress/src/main/java/org/wordpress/android/ui/sitecreation/services/SiteCreationService.kt

```diff
@@ -43,6 +45,7 @@ class SiteCreationService : AutoForeground<SiteCreationServiceState>(SiteCreatio
         val data = intent.getParcelableExtra<SiteCreationServiceData>(ARG_DATA)!!
         manager.onStart(
                 LocaleManager.getLanguageWordPressId(this),
+                localeManagerWrapper.getTimeZone().id,
```

Comment:
Since we already have a [TimeZoneProvider](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/5991f4ddf1a66799464d549bde5e7ce8c2f4e944/fluxc/src/main/java/org/wordpress/android/fluxc/utils/TimeZoneProvider.kt#L7) in FluxC uses the same Java API, I thought another idea might be to add the timezone directly from there as a default option so we're not required to specify it from outside of FluxC.

I'm leaving this question open to be discussed with the code reviewers ;)

Reply:
> I thought another idea might be to provide the timezone directly from there as a default option so we're not required to specify it from outside of FluxC.

Thanks for elaborating on your thought process here. I think your choice here is the best option because of the points you mentioned, especially with regard to flexibility, as FluxC can benefit from shared usage best when it is maintained as flexible.

I also agree that the current approach is more complex, but I don't envision that we'd gain much by using the FluxC implementation. Both implementations are simple wrappers for `TimeZone.getDefault()`, and we'd still need to expand the ABI by exposing a new method (i.e. in the current approach we do this by changing the signature, but in the alternative approach we'd presumably do this by having a new name for the new method which detects this aspect of the locale in the library).

On this point:
> Having the possibility to provide the value for `timeZoneId` from WordPress-Android while we don't have yet a specific need for this might be a sign of premature optimisation
I mostly agree, but I'd consider a candidate for premature _design_ rather than optimization - though due to the above mentioned context, it seems reasonable enough to expand the interface slightly to accommodate this flexibility, especially since time zones are an area for which affording some flexibility has a greater chance yielding a net positive return. :smiley:

Reply:
Thanks a lot for contributing to the discussion on this @mkevins and providing your view on the options and their advantages & disadvantages!

> but I don't envision that we'd gain much by using the FluxC implementation
I share your view on this, the "gain" seems to me not a big improvement in terms of handling complexity 👍 

> but I'd consider a candidate for premature design rather than optimization
This is a good alternative perspective on how to look at it 👍 

> since time zones are an area for which affording some flexibility has a greater chance yielding a net positive return
Thank you for this, it does help me weight more towards the current approach we have in the PR 🙇